### ISSUE TITLE
secure_backup_required in .well-known/matrix/client has no effect

### ISSUE THREAD

**Issue by rltas** [element-hq/element-web#26077]
### Steps to reproduce

My `https://mydomain.net/.well-known/matrix/client`:

```
{
    "im.vector.riot.e2ee": {
        "default": false
    },
    "io.element.e2ee": {
        "default": false,
        "secure_backup_required": true,
        "secure_backup_setup_methods": ["passphrase"]
    },
    "m.homeserver": {
        "base_url": "https://matrix.mydomain.net"
    },
    "org.matrix.msc3575.proxy": {
        "url": "https://matrix.mydomain.net/sliding-sync"
    }
}
```

I know it's generally working because the default encryption setting is honored.
Update: my colleague just said this works in Element on iOS, but not in Element Web.

#### What did you expect?

That I can use Element only after setting up secure backup [as per the docs](https://github.com/vector-im/element-web/blob/develop/docs/e2ee.md#requiring-secure-backup).

#### What happened instead?

No visible change.

### Application version

Element version 1.11.38, Olm version 3.2.14

### Homeserver

Synapse version 1.89.0

### Will you send logs?

No

**Comment by t3chguy** [comment]
> Will you send logs?
No

Please send logs

**Comment by rltas** [comment]
> > Will you send logs?
> > No
> 
> Please send logs

Of what exactly?

**Comment by t3chguy** [comment]
Of the app
![image](https://github.com/vector-im/element-web/assets/2403652/406a8615-7abb-485c-a9e7-06b38d67b549)

**Comment by rltas** [comment]
Dunno why I thought this was meant for something different - just sent it.

**Comment by t3chguy** [comment]
`secure_backup_required` is false in your deployment, also be careful you serve the file in 2 places.

`bXX-XXXt.XX-XX.net` & `matrix.bXX-XXXt.XX-XX.net` - the former is the only valid as per https://spec.matrix.org/v1.8/client-server-api/#well-known-uri

I suggest retrying once you ensure the former URL has `secure_backup_required=true`

**Comment by rltas** [comment]
Yeah I noticed that too, that seems to be some quirk with the ansible playbook but it's the same file being served under both urls. I changed the setting back after submitting the report, that's why it was false - I can keep it true and give you a test user if that helps.

What I just figured out:
- upon first logging in with a fresh user, no backup is enforced
- when logging in and skipping verification, no backup is enforced
- when logging in and resetting keys, backup is enforced
- when hitting F5 at that point, no backup is enforced
Is that behaviour intentional? Eventually it's how it's meant to work, but if so it's unintuitive as I can manually configure a backup anytime, starting with a fresh user, and that seemingly arbitrary flow makes it harder to on-ramp users with a step-by-step guide.
